docs/src/index.md

```diff
@@ -126,6 +126,10 @@ For translating proto files, use `protojl` function (previously `protoc`). To de
 ### Mutability
 Messages are now translated to immutable structs. This means that code that used the mutable structs to accumulate data will now have to prepare each field and construct the struct after all of them are ready.
 
+### Common abstract type
+
+By default, the generated structs don't share any common abstract type (well, except `Any`), when you set the `common_abstract_type` option to `true`, every struct definition will be a subtype of `ProtoBuf.AbstractProtoBufMessage`.
```

Comment:
Would it be more helpful to make it `common_abstract_type::Union{Nothing, Type}=nothing` and allow the _user_ to pass in the abstract type that all structs will subtype?

Reply:
That's a good point, it would allow more flexibility. What I like about the current design though is that it makes the scripts always valid (with custom abstract type, you'd have to include the generated files _after_ the abstract type was defined) and I believe it wouldn't work in "proper" proto packages (unless we want to support custom dependencies for the generated packages to import the abstract type).

Now that I think about it, I imagine that with custom abstract types one would want to apply them per message (this is an `AbstractVector`, this is a `Real`...). The issue this PR addresses wants a common abstract type because the previous implementation provided one, but besides compatibility, I don't see much use for a common type like that. I mean if we managed to get by without it when implementing the codecs, it's hard to imagine other people would really need that (again besides making the transition to 1.0 easier). But maybe we should introduce a trait people can use to detect if a given type is protobuf-serializable?

What do you think?

Reply:
Ok, that helps my understanding. I think the PR as-is is good then. If someone asks about needing a trait, then this PR provides basically that, and we can provide other functionality as requested.

Reply:
Thanks!